frontend/src/components/modals/CopyLink.vue

```diff
@@ -0,0 +1,132 @@
+          <button type="button"
+                  class="close"
+                  id="closeButton"
+                  data-toggle="tooltip"
+                  data-placement="right"
+                  title="Close"
+                  aria-label="Close"
+                  @click="close"
+          >
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          <div class="form-group">
+            <div class="input-group">
+              <input v-model="itemName"
+                     type="text"
+                     class="form-control"
+                     ref="inputField"
+                     placeholder="Link"
+              >
+            </div>
+          </div>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn font-weight-bold" id="copyButton" @click="copylink">
+            Select text
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { parseXML } from 'webdav'
+
+  export default {
+    name: 'CopyLinkModal',
+    props: ["isCopyLinkModalVisible"],
+    data() {
+      return {
+        itemName: "",
+      }
+    },
+    async updated() {
+      if (this.isCopyLinkModalVisible) {
+        let client = this.$store.getters.StateWebdavClient;
+        let path = this.$store.getters.StatePath;
+        let directoryPath = path[path.length - 1].path;
+        if (directoryPath == null) {
+            return;
+        }
+
+        let directoryClipboard = "";
+        // Checks if the selected object is in a subfolder or not
+        if (!directoryPath.endsWith("/")) {
+            // If selected object is in a subfolder, adjust path
+            directoryPath += "/";
+            // Extracts the name of the directory, also deletes last slash in string for clipboard link
+            directoryClipboard = directoryPath.split(path[0].path)[1].slice(0,-1);
+        }
+        // Builds path with directory path + fileName
+        let filePath = directoryPath + this.$store.getters.StateSelectedChildren[0].name;
+        const response = await client.customRequest(filePath, { 
+            method: "PROPFIND",
+            data: '<?xml version="1.0" encoding="UTF-8"?><d:propfind xmlns:d="DAV:"><d:prop xmlns:oc="http://owncloud.org/ns"><oc:fileid/></d:prop> </d:propfind>'
+        });
+        // Splits the response after the first fileId tag and after the second one to receive the fileId
+        const xmlData = await parseXML(response.data);
+        let fileId = xmlData.multistatus.response[0].propstat.prop.fileid;
+        // Builds link
+        this.itemName = process.env.VUE_APP_NEXTCLOUD_BASE_URL + "index.php/apps/files?dir=/"+directoryClipboard+"&openfile="+fileId;
+      }
+    },
+    methods: {
+      copylink() {
+        this.$refs.inputField.focus();
+        this.$refs.inputField.select();
+        
```

Comment:
The text is only selected, but not copied to the clipboard. I think it would make sense for the text to be copied.

Reply:
The problem is that the "copy to clipboard" function doesn't seem to work while inside ms team because of security issues. So right now the idea was to only give the user the selected link text, so that he can copy it himself. But we can test it the other way again